tests/lib/rules/no-shadow-route-definition.js

```diff
@@ -361,6 +366,35 @@ ruleTester.run('no-shadow-route-definition', rule, {
           }),
           type: 'CallExpression',
         },
+        {
+          message: buildErrorMessage({
```

Comment:
Just wondering, why is there an additional error for this case?

Reply:
```
this.route("blog", { path: "/" }, function() {
  this.route("sub-blog", { path: "/" }, function() {
    this.route("post");
   });
});
this.route("sub-blog", { path: "/" }, function() {
  this.route("post");
});
```

With previous implementation only first level conflicts where detected as errors, while actually nested routes are conflicting as well. 

In this case first conflict is:

`blog` vs `sub-blog` on top level and second conflict given this structure is `sub-blog/post` - because top level `blog` path is also just a / - therefore we have 2 routes with URL `/sub-blog/post` - regardless that one of them is nested one level deeper.

--

Change itself is driven by the fact that I changed how routeMap stores seen paths and how we check them up.